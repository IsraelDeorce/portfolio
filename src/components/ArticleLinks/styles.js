import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
