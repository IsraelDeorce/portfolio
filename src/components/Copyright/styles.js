import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
