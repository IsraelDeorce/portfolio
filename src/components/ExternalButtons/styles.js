import styled from 'styled-components'

export const ExternalButtons = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
  font-size: 1.25rem;
`
