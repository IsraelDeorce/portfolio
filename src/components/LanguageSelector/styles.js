import styled from 'styled-components'

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: ${({ theme }) => theme.c_brackets};
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`