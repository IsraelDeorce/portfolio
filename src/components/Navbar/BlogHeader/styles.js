import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
