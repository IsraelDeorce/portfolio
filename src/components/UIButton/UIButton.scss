.ui-button {
  align-self: stretch;
  display: flex;
  justify-content: center;
  padding: .75rem 1.25rem;
}

.ui-button-label {
  padding: 0 0.25rem;
}