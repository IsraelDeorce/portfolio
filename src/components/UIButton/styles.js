import styled, { css } from 'styled-components'

export const UIButton = styled.button`
  display: flex;
  justify-content: center;
  padding: .75rem 1.25rem;
  cursor: pointer;
  > *:not(:last-child) {
    padding-right: 0.25em;
  }
  ${({ theme, secondary }) => secondary
  ? css`
    color: ${theme.c_component};
  `
  : css`
    background-color: ${theme.c_component};
    color: ${theme.c_background};
  `}
`
