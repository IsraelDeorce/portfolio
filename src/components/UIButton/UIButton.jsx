import PropTypes from 'prop-types'
import * as S from './styles'

const propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func,
  leftIcon: PropTypes.element,
  rightIcon: PropTypes.element,
  secondary: PropTypes.bool
}

function UIButton(props) {
  const {
    label = '',
    onClick = () => alert('I do nothing!'),
    leftIcon = null,
    rightIcon = null,
    secondary = false
  } = props

  return (
    <S.UIButton onClick={onClick} secondary={secondary}>
      {leftIcon}
      <span>{label}</span>
      {rightIcon}
    </S.UIButton>
  )
}

UIButton.propTypes = propTypes

export default UIButton
