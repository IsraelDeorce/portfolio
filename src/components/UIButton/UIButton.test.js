import UIButton from './UIButton'
import * as S from './styles'

describe('<UIButton/>', () => {
  it('it renders a button with left icon, a label and a right icon', () => {
    const Element = (props) => <div>{props.desc}</div>
    const leftIcon = <Element desc='leftIcon' />
    const rightIcon = <Element desc='rightIcon' />
    const label = 'Lorem Ipsum'
    const shallowed = shallow(<UIButton leftIcon={leftIcon} rightIcon={rightIcon} label={label} />)
    expect(shallowed.find(S.UIButton).children()).toHaveLength(3)
    expect(shallowed.find(S.UIButton).childAt(0).find('Element[desc="leftIcon"]').exists()).toBe(true)
    expect(shallowed.find(S.UIButton).childAt(1).find('span').text()).toBe(label)
    expect(shallowed.find(S.UIButton).childAt(2).find('Element[desc="rightIcon"]').exists()).toBe(true)
  })

  it('sets the secondary theme', () => {
    expect(shallow(<UIButton secondary />).find(S.UIButton).prop('secondary')).toBe(true)
  })

  it('calls the onClick function when it is passed', () => {
    const mockFn = jest.fn()
    shallow(<UIButton onClick={mockFn} />).find(S.UIButton).simulate('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<UIButton />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<UIButton />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<UIButton />)).toMatchSnapshot()
    })
  })
})
