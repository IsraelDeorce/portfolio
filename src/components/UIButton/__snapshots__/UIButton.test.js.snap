// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UIButton/> Snapshot renders without crashing using mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: .75rem 1.25rem;
  cursor: pointer;
}

.c0 > *:not(:last-child) {
  padding-right: 0.25em;
}

<UIButton>
  <styled.button
    onClick={[Function]}
    secondary={false}
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      <span />
    </button>
  </styled.button>
</UIButton>
`;

exports[`<UIButton/> Snapshot renders without crashing using render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: .75rem 1.25rem;
  cursor: pointer;
}

.c0 > *:not(:last-child) {
  padding-right: 0.25em;
}

<button
  class="c0"
>
  <span />
</button>
`;

exports[`<UIButton/> Snapshot renders without crashing using shallow 1`] = `
<styled.button
  onClick={[Function]}
  secondary={false}
>
  <span />
</styled.button>
`;
